src/main/java/com/android/volley/toolbox/CacheHeader.java

```diff
@@ -143,4 +145,31 @@ boolean writeHeader(OutputStream os) {
             return false;
         }
     }
+
+    /** Writes the contents of this CacheHeader to the specified ByteBuffer. */
+    void writeHeader(ByteBuffer buffer) throws IOException {
+        buffer.putInt(CACHE_MAGIC);
+        DiskBasedCacheUtility.writeString(buffer, key);
+        DiskBasedCacheUtility.writeString(buffer, etag);
+        buffer.putLong(serverDate);
+        buffer.putLong(lastModified);
+        buffer.putLong(ttl);
+        buffer.putLong(softTtl);
+        DiskBasedCacheUtility.writeHeaderList(allResponseHeaders, buffer);
+    }
+
+    /** Gets the size of the header in bytes */
+    int getHeaderSize() throws IOException {
+        int x = 44; // 4 longs, 3 ints
+        try {
+            x += key.getBytes("UTF8").length;
+            if (etag != null) {
+                x += etag.getBytes("UTF8").length;
+            }
+        } catch (UnsupportedEncodingException e) {
+            VolleyLog.e(e.toString());
```

Comment:
Could you log the exception with a message instead?

```java
VolleyLog.e(e, "Your message here")
```

Reply:
fixed